### DISCUSSION TITLE
mGBA Runner

### DISCUSSION THREAD

**Discussion by ghost** [lutris/lutris#5028] [Ideas]
I noticed there isn't a non-Libretro runner for GBA games. mGBA is a solid GBA emulator. I don't know if the license is compatible with Lutris using it as a runner, it is licensed under the MPL 2.0.

**Comment by strycore** [comment]
We don't have any particular requirements regarding licenses. The biggest reason why we don't have a dedicated GBA emulator is that they're available through Retroarch. 
If there are advantages to running mGBA standalone then we would accept a runner for it, especially if it's simple and can be written in JSON instead of Python.

**Comment by ghost** [comment]
Thanks you! I asked to make sure it's okay with the mGBA devs as well. https://github.com/mgba-emu/mgba/issues/3025
I've noticed RetroArch can lag behind in performance optimizations. There's also some users that prefer to avoid libretro due to some of their past decisions. I'll make a pull request assuming the developers approve.

**Comment by strycore** [comment]
We have never asked for upstream approval for any of our runners and this is not something that is needed, especially when the software is open source.

**Comment by ghost** [comment]
On Wed, 2023-09-20 at 16:04 -0700, Mathieu Comandon wrote:
> We have never asked for upstream approval for any of our runners and
> this is not something that is needed, especially when the software is
> open source. 
> 
Oh okay

**Comment by Testsr** [comment]
For the MPL, you can use the same requirements as for the GPLv2+ dolphin emulator. The licences are similar enough for the purpose of binary distribution.

**Comment by ghost** [comment]
I made the pull request: https://github.com/lutris/lutris/pull/5030
